app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/item/MessageItemComponents.kt

```diff
@@ -274,14 +280,37 @@ internal fun MessageDecryptionFailure(
 
 @Composable
 internal fun Modifier.customizeMessageBackground(
+    defaultBackgroundColor: Color,
     sendingFailed: Boolean,
-    receivingFailed: Boolean
-) = run {
-    if (sendingFailed || receivingFailed) {
-        background(MaterialTheme.wireColorScheme.messageErrorBackgroundColor)
-    } else {
-        this
-    }
+    receivingFailed: Boolean,
+    isDeleted: Boolean,
+    isSelectedMessage: Boolean,
+    selfDeletionTimerState: SelfDeletionTimerHelper.SelfDeletionTimerState,
+    colorAnimation: Color
+): Modifier {
+
+    return this
+        .then(if (isSelectedMessage) drawBehind { drawRect(colorAnimation) } else this)
+        .then(
+            when {
+                sendingFailed || receivingFailed -> {
+                    background(MaterialTheme.wireColorScheme.messageErrorBackgroundColor)
+                }
+
+                selfDeletionTimerState is SelfDeletionTimerHelper.SelfDeletionTimerState.Expirable && !isDeleted -> {
+                    val color by animateColorAsState(
```

Comment:
question: this is also actually an animation, so it's a bit unclear why we have `colorAnimation` parameter but here we have an animation created inside 😅 can't we unify that?